File: testdata/fixtures/ai/typescript/detect-gemini.ts
// AI usage detection: Gemini
// ruleid: detect-gemini
import "@google/generative-ai";
// ruleid: detect-gemini
import { GoogleGenerativeAI } from "@google/generative-ai";
// ruleid: detect-gemini
const genAI = new GoogleGenerativeAI("key");
// ruleid: detect-gemini
const model = genAI.getGenerativeModel({ model: "gemini-pro" });
// ok: detect-gemini
import fs from "fs";
// ok: detect-gemini
const text = "plain string";
// ok: detect-gemini
console.log("hello");

File: testdata/fixtures/ai/swift/detect-gemini.swift
// AI usage detection: Gemini
// ruleid: detect-gemini
import GoogleGenerativeAI
import Foundation
// ruleid: detect-gemini
let model = GenerativeModel(name: "gemini-pro", apiKey: "key")
// ruleid: detect-gemini
let other = GenerativeModel(name: "gemini-flash", apiKey: "key")
// ok: detect-gemini
import Dispatch
// ok: detect-gemini
let text = "plain string"
// ok: detect-gemini
let x = 1
